### PR TITLE
Reenable TS eslint checks

### PR DESCRIPTION
**Pull request by nbeloglazov**
### Description

Reenables a bunch of TS checks so to improve code quality. To reenable the checks had to fix code here and there. No major changes. One of the checks helped to catch non-working test method in ti_dashboard.ts that did nothing while it claimed to assert DOM. 

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/application.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://docs.civiform.us/contributor-guide/developer-guide/feature-flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### CODE REVIEW

File: server/app/assets/javascripts/admin_programs.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
+// !!! DO NOT SUBMIT ME
```
Comment (shanemc-goog) [inline server/app/assets/javascripts/admin_programs.ts]: ping

**Review by shanemc-goog — CHANGES_REQUESTED** [review]

File: browser-test/src/support/ti_dashboard.ts [hunk 2]
```diff
@@ -33,19 +33,20 @@ export class TIDashboard {
...
     expect(rowText).toContain(client.lastName)
+    // date of birth rendered as <input> rather than plain text.
+    expect(await row.locator('input[name="dob"]').inputValue()).toEqual(
+      client.dobDate,
+    )
   }
 
   async expectDashboardNotContainClient(client: ClientInformation) {
     const tableInnerText = await this.page.innerText('table')
-    expect(tableInnerText).not.toContain(this.convertToMMDDYYYY(client.dobDate))
```
Comment (shanemc-goog) [inline browser-test/src/support/ti_dashboard.ts:48]: Can this check not be maintained?

**Review by shanemc-goog — CHANGES_REQUESTED** [review]

File: browser-test/src/support/ti_dashboard.ts (hunk 2 above)
Comment (nbeloglazov) [reply browser-test/src/support/ti_dashboard.ts:48]: Not easily. We need to check that table has no <input> fields with value equal to the date of birth. It gets very specific and britle. Checking that table doesn't contain email and name should be sufficient to ensure that client doesn't is not rendered.

File: server/app/assets/javascripts/admin_programs.ts (hunk 1 above)
Comment (nbeloglazov) [reply server/app/assets/javascripts/admin_programs.ts]: Fixed in latest merge.

**Review by shanemc-goog — APPROVED** [review]

**Pull request merged**